src/main/java/teammates/ui/webapi/CreateAccountRequestAction.java

```diff
@@ -1,56 +1,198 @@
-        String instructorInstitution = createRequest.getInstructorInstitution().trim();
+        String instructorHomePageUrl = createRequest.getInstructorHomePageUrl().trim();
+        String comments = createRequest.getComments().trim();
 
-        AccountRequestAttributes accountRequestToCreate = AccountRequestAttributes
-                .builder(instructorEmail, instructorInstitution, instructorName)
-                .build();
+        AccountRequestAttributes accountRequestToCreate;
         AccountRequestAttributes accountRequestAttributes;
 
-        try {
-            accountRequestAttributes = logic.createAccountRequest(accountRequestToCreate);
-            // only schedule for search indexing if account request created successfully
-            taskQueuer.scheduleAccountRequestForSearchIndexing(instructorEmail, instructorInstitution);
-        } catch (InvalidParametersException ipe) {
-            throw new InvalidHttpRequestBodyException(ipe);
-        } catch (EntityAlreadyExistsException eaee) {
-            // Use existing account request
-            accountRequestAttributes = logic.getAccountRequest(instructorEmail, instructorInstitution);
+        switch (intent) {
+        case ADMIN_CREATE:
+            accountRequestToCreate = AccountRequestAttributes
+                    .builder(instructorName, instructorInstitute, instructorEmail, instructorHomePageUrl, comments)
+                    .build();
+
+            try {
+                accountRequestAttributes = logic.createAndApproveAccountRequest(accountRequestToCreate);
+                // only schedule for search indexing if account request created successfully
+                taskQueuer.scheduleAccountRequestForSearchIndexing(accountRequestAttributes.getEmail(),
+                        accountRequestAttributes.getInstitute());
+            } catch (EntityAlreadyExistsException eaee) {
+                AccountRequestAttributes accountRequest = logic.getAccountRequest(instructorEmail, instructorInstitute);
+                throw new InvalidOperationException(
+                        "An account request already exists with status " + accountRequest.getStatus() + ".", eaee);
+            } catch (InvalidParametersException ipe) {
+                throw new InvalidHttpRequestBodyException(ipe);
+            }
+
+            String joinLink = accountRequestAttributes.getRegistrationUrl();
+            EmailWrapper joinEmail = emailGenerator.generateNewInstructorAccountJoinEmail(
+                    instructorEmail, instructorName, joinLink);
+            emailSender.sendEmail(joinEmail);
+
+            return new JsonResult(new JoinLinkData(joinLink));
+
+        case PUBLIC_CREATE:
+            AccountRequestCreateErrorResults errorResults = new AccountRequestCreateErrorResults();
+            if (!validateAccountRequestAndPopulateErrorResults(instructorName, instructorInstitute, instructorCountry,
```

Comment:
As I mentioned earlier, there should be no need to pass institute and country to the backend. 

1) Avoid having two separate definitions for institute here. This is extra complexity that is not obvious and is a potential source of future bugs.
2) The previous flow of sending the combined institute works perfectly well here.
3) There is no need to validate either the "pure institute" or "pure country" in the backend.
   - As per status quo the backend does not understand the concept of separate country and institute. This is not used in the logic. Any future needs are irrelevant to this discussion. This can be added when needed, it is not needed right now.
   - There are no practical validation rules for this at present. There is no real reason to validate these on top of validation that already exists for institute.
   - Validation for complex rules is done manually right now. We might decide to implement automated checks in the future, but that is an issue not relevant to this PR.

Reply:
The previous approach was fine because it's only for admin and admin combines the institute and country directly and entered them into one input box, but this is no longer the case with the new input form.